src/Foundatio.RabbitMQ/Messaging/RabbitMQMessageBus.cs

```diff
@@ -117,21 +131,22 @@ public class RabbitMQMessageBus : MessageBusBase, IMessageBus {
                 return;
             _logger.Trace("Message Publish: {messageType}", messageType.FullName);
 
-            // RabbitMQ only supports delayed messages with a third party plugin.
-            if (delay.HasValue && delay.Value > TimeSpan.Zero) {
-                await AddDelayedMessageAsync(messageType, message, delay.Value).AnyContext();
-                return;
-            }
-
             var data = await _serializer.SerializeAsync(new MessageBusData {
                 Type = messageType.AssemblyQualifiedName,
                 Data = await _serializer.SerializeToStringAsync(message).AnyContext()
             }).AnyContext();
             
             var basicProperties = _publisherChannel.CreateBasicProperties();
             basicProperties.Persistent = _persistent;
-            basicProperties.Expiration = _defaultMessageTimeToLive.Milliseconds.ToString();
-
+            basicProperties.Expiration = _defaultMessageTimeToLive.TotalMilliseconds.ToString(CultureInfo.InvariantCulture);
+            // RabbitMQ only supports delayed messages with a third party plugin called "rabbitmq_delayed_message_exchange"
+            if (_delayedExchange && delay.HasValue && delay.Value > TimeSpan.Zero) {
```

Comment:
When this exception occurs we could store it in a flag, and then short circuit it if not available so it only happens once.